soh/soh/Enhancements/timesaver_hook_handlers.cpp

```diff
@@ -720,6 +721,18 @@ void TimeSaverOnVanillaBehaviorHandler(GIVanillaBehavior id, bool* should, va_li
 
             break;
         }
+        case VB_PLAY_GATE_OPENING_OR_CLOSING_CS: {
+            if (CVarGetInteger(CVAR_ENHANCEMENT("TimeSavers.SkipMiscInteractions"), IS_RANDO)) {
+                EnHeishi2* enHeishi2 = va_arg(args, EnHeishi2*);
+                enHeishi2->unk_2F2[0] = 0;
+
+                // The second argument determines whether the vanilla code should be run anyway. It
+                // should be set to `true` ONLY IF said code calls `Play_ClearCamera`, false otherwise.
+                bool clearCamera = va_arg(args, bool);
```

Comment:
```suggestion
                bool clearCamera = (bool)va_arg(args, int);
```
Bools passed in via args are turned into ints and need converting back after va_arg to avoid the crash